### ISSUE TITLE
After reinstalling Windows 10 Version 1703 Build 15063 from Backup uBlock doesnt reinstallable

### ISSUE THREAD

**Issue by hrodeberht** [nikrolls/uBlock-Edge#84]
uBlock worked fine until I needed to reinstall Windows 10 Pro Version 1703 Build 15063 from full backup. I lost Windows Activation and I needed to reactivate it. I didn't saw that that at first moment and started Microsoft Edge. I lost the licences for store apps too and  uBlock got deinstalled/deactivated? not sure.

After reactivation of Windows I started Microsoft Edge but uBlock-Icon disappeared from Edge. Under Settings -> Extensions it disappeared too. I selected the function too install a Extension in Edge. Searched for uBlock. The Message was (translated from german). "The extension has caused a mistake and must be newly installed for error correction . 

From Store I reinstalled uBlock, but the same above message appears again and again.

I followed this guide (in german):
https://www.deskmodder.de/wiki/index.php/Microsoft_Edge_reparieren_neu_installieren
Deletion of the Profile Folder/Cache/Registry Keys from Edge and so on, but I never rebooted and it didn't helped.

This guide similar to the next one, doesnt helped too:
https://answers.microsoft.com/en-us/windows/forum/windows_10-windows_store/unable-to-install-extension-for-microsoft-edge/68870e51-1fe7-4665-8c88-898c3abfc374
Post from marsu5. Deinstalled uBlock, Reinstall of all stock apps/Microsoft Store.

Last try was to follow this tutorial as I found in another closed issues. I followed exactly as described and rebooted twice.
https://www.windowscentral.com/how-to-reset-microsoft-edge-windows-10

But it didn't helped too, what can I do? Which information I can provide you.

Best Regards
hrodeberht

**Comment by hrodeberht** [comment]
Version of MS Edge: 40.15063.0.0

**Comment by sgeto** [comment]
Hey this is just a wild guess. If resetting edge doesn't help then maybe you should try to clear the windows store cache?

http://www.thewindowsclub.com/windows-store-cache-may-damaged
http://windowsitpro.com/windows-81/three-options-fixing-windows-store-problems

viel gluck

**Comment by hrodeberht** [comment]
Hi sgeto,

The diagnostic tool complained that I have no microsoft account and I had to register for one, what I didn't do. Errors, like windows store cache is damaged didn't showed up, but I followed your links/tips and combined them with maximum brutality :-): Before I was able to install other Apps/Microsoft Edge Extensions from Store, but just not uBlock.

I tried yet a second time and documented what I have done:

deletion of the complete Folder Content from:
C:\Users\<USERNAME>\AppData\Local\Packages\Microsoft.MicrosoftEdge_8wekyb3d8bbwe

Delete following content of following registry path:
HKEY_CURRENT_USER\SOFTWARE\Classes\Local Settings\Software\Microsoft\Windows\CurrentVersion\AppContainer\Storage\microsoft.microsoftedge_8wekyb3d8bbwe\MicrosoftEdge

Reregistrated Microsoft App Package with Microsoft Powershell (as admin)
Get-AppXPackage -AllUsers -Name Microsoft.MicrosoftEdge | Foreach {Add-AppxPackage -DisableDevelopmentMode -Register “$($_.InstallLocation)\AppXManifest.xml” -Verbose}

Reboot

Windows Powershell (Processing showed up and succesfull removed uBlock)
Get-AppxPackage *uBlockOrigin* | Remove-AppxPackage

If I can trust the tutorial this will reinstall the complete store with all apps:
Get-AppxPackage -AllUsers| Foreach {Add-AppxPackage -DisableDevelopmentMode -Register “$($_.InstallLocation)\AppXManifest.xml”}

As sgeto advised me. I think this do the trick, but I have completely documented what I have done, possible it can help someone other:

Reset Windows Store Cache (again: explorer.exe as administrator)
C:\Windows\System32\wsreset.exe

Explorer.exe (as admin) as advised tried deleting the cache folder. The Folder localState was in my case empty.
C:\Users\<USERNAME>\AppData\Local\Packages\Microsoft.WindowsStore_8wekyb3d8bbwe\LocalState

Reboot

Searched for uBlock Origin and installed it from Windows Store. Sucessfull activated and uBlock started with no error messages anymore.

Thank you very much for your help.

Im new here have I to close the thread as solved or would this be done from the package maintainer?

Best Regards
hrodeberht
